pyparsing/unicode.py

```diff
@@ -102,14 +102,16 @@ def identbodychars(cls):
         all characters in this range that are valid identifier body characters,
         plus the digits 0-9, and · (Unicode MIDDLE DOT)
         """
+        identifier_chars = (
+            c for c in cls._chars_for_ranges
+            if ("_" + c).isidentifier()
+        )
         return "".join(
             sorted(
                 set(
                     cls.identchars
                     + "0123456789·"
-                    + "".join(
-                        [c for c in cls._chars_for_ranges if ("_" + c).isidentifier()]
-                    )
+                    + "".join(identifier_chars)
```

Comment:
You could just remove the []'s in line 111 and achieve the same result. IIRC, I intentionally made this a list comprehension instead of just a generator expression, since "".join can pre-allocate space when passed a list, but can't when passed a generator.

But since the whole reason we are building this internal string is to build a set, maybe the solution is to make this its own set using the generator you created or just a genexp, and '|' it with the smaller `set(cls.identchars + "0123456789·")`. Keep your change, but make `identifier_chars` a set.